### PR TITLE
557 - python api policies test

### PR DESCRIPTION
**Pull request by manrodrigues**

### CODE REVIEW

File: python-test/features/steps/control_plane_policies.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    """
+    token = context.token
+    policies_list = list_policies(token)
+    policies_filtered_list = filter_list_by_parameter_start_with(policies_list, 'name', policy_name_prefix)
+    delete_policies(token, policies_filtered_list)
+
+
+# Todo complete the documentation
+def create_policy(token, policy_name, handler_label, handler, description=None, tap="default_pcap",
+                  input_tap="pcap", host_specification=None, filter_expression=None, backend_type="pktvisor"):
```
Comment (stefano-daitan) [inline python-test/features/steps/control_plane_policies.py:39]: Typo: `input_tap` -> `input_type`;

File: python-test/features/policies.feature [hunk 2]
```diff
@@ -0,0 +1,8 @@
+@policies
+Feature: policy creation
+
+  Scenario: Create a policy
+    Given the Orb user logs in
+      And that an agent already exists and is online
+    When a new policy is created
+    Then referred policy must be listened on the orb policies list
```
Comment (stefano-daitan) [inline python-test/features/policies.feature:8]: Should we also check in the orb agent if the creation of this policy had effect?

**Review by stefano-daitan — COMMENTED** [review]
Good job, just a few questions

File: python-test/features/steps/control_plane_policies.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+@when("a new policy is created")
+def create_new_policy(context):
+    context.policy = create_policy(context.token, name, handle_label, handler)
+
+
+@then("referred policy must be listened on the orb policies list")
+def check_policies(context):
+    policy_id = context.policy['id']
+    get_policy(context.token, policy_id)
```
[the remark this answers is not in the document]
Comment (manrodrigues) [reply python-test/features/steps/control_plane_policies.py]: We already have an assert in the get function.. I think it's enough for now

File: python-test/features/steps/control_plane_policies.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
+from hamcrest import *
+import requests
+from control_plane_agents import base_orb_url
+from behave import given, when, then
+from utils import random_string, filter_list_by_parameter_start_with
+
+policy_name_prefix = "test_policy_name"
```
[the remark this answers is not in the document]
Comment (manrodrigues) [reply python-test/features/steps/control_plane_policies.py]: yes

File: python-test/features/policies.feature (hunk 2 above)
Comment (manrodrigues) [reply python-test/features/policies.feature:8]: we will only be able to see the effects in the orb agent after creating the dataset

**Comment by stefano-daitan** [comment]
@manrodrigues I'll wait for you to finish the docs before approving this PR.

File: python-test/features/steps/control_plane_policies.py [hunk 5]
```diff
@@ -35,13 +35,17 @@ def clean_policies(context):
...
                   input_type="pcap", host_specification=None, filter_expression=None, backend_type="pktvisor"):
     """
 
     Creates a new policy in Orb control plane
 
+    :param filter_expression:
+    :param (str) host_specification: Subnets (comma separated) which should be considered belonging to this host,
+    in CIDR form. Used for ingress/egress determination, defaults to host attached to the network interface.
+    :param input_type: this must reference a tap name, or application of the policy will fail
+    :param tap: named, host specific connection specifications for the raw input streams accessed by pktvisor
```
Comment (stefano-daitan) [inline python-test/features/steps/control_plane_policies.py:48]: Ideally the parameters are documented in the same order as they are declared in the function. But let's merge this PR and fix this in #597

**Review by stefano-daitan — APPROVED** [review]

File: python-test/features/steps/control_plane_policies.py (hunk 5 above)
Comment (stefano-daitan) [reply python-test/features/steps/control_plane_policies.py:48]: @manrodrigues please resolve this conversation when you're ready to merge this PR.

Comment (manrodrigues) [reply python-test/features/steps/control_plane_policies.py:48]: ok.

**Pull request merged**